Nodejs/Product/Analysis/Analysis/AnalysisLimits.cs

```diff
@@ -34,6 +34,17 @@ internal class AnalysisLimits {
             MaxObjectLiteralProperties = 50;
             MaxObjectKeysTypes = 5;
             MaxMergeTypes = 5;
+            NestedModulesLimit = 4;
```

Comment:
Reason for 4 following - I try to be very practical about numbers. 

For small applications this number more then enough, and they don't have any issues dealing with some potential errors in Intellisense.

But if we have large application with Full setting, then VS Tools for Node.JS as a product immediately became unusable for that users.

I try to imagine dependency tree which exceed this 4 depth limit. Here the examples of trees about which I thought:
## Example 1
Level 0 - Large system. Some code should use properties of the object on level 4 here. 
Level 1 - Subsystem - pass data from level 4
Level 2 - Internal dependency for company - Do some work and pass data to level 1
Level 3 - Framework on top of which created Internal dependency.
Level 4 - Dependency of the framework. Objects from here still should be available.
Level 5 - Dependency of the framework provide some usefull primitive which would be used very often on the level of whole system. Hmmm. 

## Example 2 (reason why I could increase to 5)
Level 0 - Large system. Some code should use properties of the object on level 4 here. 
Level 1 - Subsystem - pass data from level 4
Level 2 - Internal dependency for company - Wrap access to internal library and perform business logic. Do some work and pass data to level 1
Level 3 - Internal library which wrap access to API.
Level 4 - Http library.
Level 5 - Promise Polyfill. 

All these examples are highly speculative and I specifically try to create such deep level. If you develop on windows with such deep level you already close to your limit, your maximum is probably 10. 
If you have such deep structure already, chances that you have more then one such deep dependency is very high, and thus we just increase time for loading solution, loading analysis file and increase memory consumption without any practical benefits.

From my point of view practical benefits much better, and better have good and usable product with 90% working intellisense.
Or we just have 2 options: Medium and None without Full being usable by anyone on the large projects.